apps/server/test/modules/task/submission.e2e-spec.ts

```diff
@@ -37,128 +37,250 @@ class API {
+		const response = await request(this.app.getHttpServer())
+			.delete(`/submissions/${submissionId}`)
+			.set('Accept', 'application/json')
+			.set('Authorization', 'jwt');
+
+		return {
+			status: response.status,
+			error: response.body as ApiValidationError,
+			result: response.body as SubmissionStatusListResponse,
+		};
+	}
 }
 
 describe('Submission Controller (e2e)', () => {
-	let app: INestApplication;
-	let currentUser: ICurrentUser;
-	let api: API;
-	let em: EntityManager;
-
-	beforeAll(async () => {
-		const module: TestingModule = await Test.createTestingModule({
-			imports: [ServerTestModule],
-		})
-			.overrideGuard(JwtAuthGuard)
-			.useValue({
-				canActivate(context: ExecutionContext) {
-					const req: Request = context.switchToHttp().getRequest();
-					req.user = currentUser;
-
-					return true;
-				},
+	describe('find statuses by task', () => {
+		let app: INestApplication;
+		let currentUser: ICurrentUser;
+		let api: API;
+		let em: EntityManager;
+
+		beforeAll(async () => {
+			const module: TestingModule = await Test.createTestingModule({
+				imports: [ServerTestModule],
 			})
-			.compile();
+				.overrideGuard(JwtAuthGuard)
+				.useValue({
+					canActivate(context: ExecutionContext) {
+						const req: Request = context.switchToHttp().getRequest();
+						req.user = currentUser;
 
-		app = module.createNestApplication();
-		await app.init();
-		api = new API(app);
-		em = module.get(EntityManager);
-	});
+						return true;
+					},
+				})
+				.compile();
 
-	beforeEach(async () => {
-		await cleanupCollections(em);
-	});
+			app = module.createNestApplication();
+			await app.init();
+			api = new API(app);
+			em = module.get(EntityManager);
+		});
 
-	afterAll(async () => {
-		await app.close();
-	});
+		beforeEach(async () => {
+			await cleanupCollections(em);
+		});
 
-	describe('WHEN user is not authorized', () => {
-		it('should return 401', async () => {
-			const taskId = 'id';
+		afterAll(async () => {
+			await app.close();
+		});
 
-			const { status } = await api.findStatusesByTask(taskId);
+		describe('WHEN user is not authenticated', () => {
+			it('should return 401', async () => {
+				const taskId = 'id';
 
-			expect(status).toEqual(401);
-		});
-	});
+				const { status } = await api.findStatusesByTask(taskId);
 
-	describe('WHEN user is authorized, has permission', () => {
-		const setup = async () => {
-			const roles = roleFactory.buildList(1, {
-				permissions: [Permission.SUBMISSIONS_VIEW],
+				expect(status).toEqual(401);
 			});
-			const user = userFactory.buildWithId({ roles });
-			const task = taskFactory.buildWithId();
-			const courseGroup = courseGroupFactory.buildWithId();
-			const submission = submissionFactory.buildWithId({ task, student: user, grade: 97, courseGroup });
+		});
 
-			await em.persistAndFlush([submission]);
-			em.clear();
-			currentUser = mapUserToCurrentUser(user);
+		describe('WHEN user is authorized, has permission', () => {
```

Comment:
Accordingly to the above this should be "user is autheticated and has permission" or "user is authenticated and authorized".